var/spack/repos/builtin/packages/py-esmpy/package.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyEsmpy(PythonPackage):
+    """ESMPy is a Python interface to the Earth System Modeling Framework (ESMF)"""
+
+    homepage = "https://earthsystemmodeling.org/"
+    git = "https://github.com/esmf-org/esmf.git"
+
+    # because it relies on py-setuptools-git-versioning, we need to install from git
+    version("develop", branch="develop")
+    version("8.4.0", sha256="28531810bf1ae78646cda6494a53d455d194400f19dccd13d6361871de42ed0f")
+    version("8.3.1", sha256="6c39261e55dcdf9781cdfa344417b9606f7f961889d5ec626150f992f04f146d")
+    version("8.3.0", sha256="0ff43ede83d1ac6beabd3d5e2a646f7574174b28a48d1b9f2c318a054ba268fd")
+    version("8.2.0", sha256="3693987aba2c8ae8af67a0e222bea4099a48afe09b8d3d334106f9d7fc311485")
+    version("8.1.1", sha256="58c2e739356f21a1b32673aa17a713d3c4af9d45d572f4ba9168c357d586dc75")
+    version("8.0.1", sha256="9172fb73f3fe95c8188d889ee72fdadb4f978b1d969e1d8e401e8d106def1d84")
+    version("8.0.0", sha256="051dca45f9803d7e415c0ea146df15ce487fb55f0fce18ca61d96d4dba0c8774")
+
+    maintainers("angus-g")
+
+    # require an in-sync version of ESMF
+    for v in ["develop", "8.4.0", "8.3.1", "8.3.0", "8.2.0", "8.1.1", "8.0.1", "8.0.0"]:
+        depends_on(f"esmf@{v}", when=f"@{v}", type=("build", "run"))
+
+    depends_on("py-setuptools@41:", type="build")
+    depends_on("py-build", when="@:8.3.1", type="build")
+    depends_on("py-setuptools-git-versioning", when="@8.4.0:", type="build")
+    depends_on("py-numpy", type="run")
```

Comment:
```suggestion
    depends_on("py-numpy", type=("build", "run"))
```

Reply:
Also missing dep on importlib-metadata and pytest-json-report

Reply:
I'm sure it's only used for testing, but it's listed in `pyproject.toml` as a required dependency. Although we currently disable checks for dependencies during the install, that's something I would like to be able to fix someday. So it's best to add it as a dependency like it's documented, or to patch it and remove it. If you do patch it, I would open a PR with esmf to fix this.

Reply:
Ah of course, I was looking at the git version where it had been moved to an optional dependency already. I've just added that into this PR. I can move history around to get the commits in the right order and have one for each of py-esmpy and py-pytest-json-report if needed.